### PR TITLE
feat: support tablet auth and flag to disable auth

### PR DESCRIPTION
**Pull request by oh2024**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
1. Add username/password authentication to tablet servers
2. Added `skip_grant_tables` flag to disable auth

* **What is the current behavior?** (You can also link to an open issue here)
Tablet servers have only service-auth enabled, trusting the sdefault token. 

* **What is the new behavior (if this is a feature change)?**
Tablet servers will use the username and password from the user table for authentication

### CODE REVIEW

File: src/flags.cc [hunk 1]
```diff
@@ -186,3 +186,4 @@ DEFINE_uint32(keep_log_file_num, 5, "Maximal info log files to be kept");
 DEFINE_int32(sync_job_timeout, 30 * 60 * 1000,
              "sync job timeout, unit is milliseconds, should <= server.channel_keep_alive_time in TaskManager");
 DEFINE_int32(deploy_job_max_wait_time_ms, 30 * 60 * 1000, "the max wait time of waiting deploy job");
+DEFINE_bool(skip_grant_tables, true, "skip the grant tables");
```
Comment (aceforeverd) [inline src/flags.cc:189]: IMO `no_auth` or `skip_auth` better

**Review by aceforeverd — CHANGES_REQUESTED** [review]

File: src/flags.cc (hunk 1 above)
Comment (oh2024) [reply src/flags.cc:189]: MySQL uses the same name

**Review by vagetablechicken — CHANGES_REQUESTED** [review]
add doc:
- how to use flag in server side, which servers, need restart
- how to use in client side, which clients, need restart

Release test should follow the steps in doc.

File: src/flags.cc (hunk 1 above)
Comment (tobegit3hub) [reply src/flags.cc:189]: It may be ok to use `skip_grant_tables` which is compatible with mysql.

**Comment by vagetablechicken** [comment]
> add doc:
> 
> * how to use flag in server side, which servers, need restart
> * how to use in client side, which clients, need restart
> 
> Release test should follow the steps in doc.

是不是所有server和client（java python cli）都能用这个开关
这个开关是090才有，更早的版本不要用。
这个开关打开要重启生效。
server不打开，client打开是什么样。部分server打开，是什么样。
认证错误，客户端收到的错误是什么样的

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: docs/zh/deploy/auth.md [hunk 2]
```diff
@@ -12,6 +12,9 @@ OpenMLDB 0.8.4版本及以后，可以使用有密码的ZooKeeper。需要在启
 
 ## OpenMLDB身份认证
 
+### 0.9.0 中的 Alpha 功能
+OpenMLDB 在 0.9.0 版本中作为 alpha 功能引入了服务器端身份验证。 从此版本开始，默认情况下不启用服务器端身份验证。 这种方法可确保向后兼容性并最大限度地降低破坏现有部署的风险。 用户可以通过在启动客户端和服务器时配置“--noskip_grant_tables”来启用服务端认证。 禁用身份验证时，不支持更改身份验证设置或修改用户凭据的语句，例如“CREATE USER”、“ALTER USER”和“DELETE USER”。
+
 OpenMLDB 0.8.5版本及以后，集群内可以使用用户名和密码进行身份认证。自定义用户和密码的创建都在集群启动之后，初始集群只有root用户，密码为空。默认情况下，任何的server和client都是以root用户名和空密码连接到集群。
```
Comment (vagetablechicken) [inline docs/zh/deploy/auth.md]: 0.8.5可使用的客户端认证，单独做一章，它不是0.9.0这一章的内容

**Review by vagetablechicken — APPROVED** [review]

**Pull request merged**